added sections and references in credentials handling in ECK (#2366)
elastic/docs-content: deploy-manage/users-roles/cluster-or-deployment-auth/managed-credentials-eck.md

::::{warning}
This command regenerates auto-generated credentials of **all** {{stack}} applications in the namespace.
::::

## Creating custom users

### Native realm

You can create custom users in the {{es}} native realm using {{es}} user management APIs or {{kib}}.

Refer to [Native user authentication](/deploy-manage/users-roles/cluster-or-deployment-auth/native.md) for more details.

### File realm

Custom users can also be created by providing the desired file realm content or a username and password in Kubernetes secrets, referenced in the {{es}} resource.

> Refer to [File-based user authentication](/deploy-manage/users-roles/cluster-or-deployment-auth/file-based.md) for extra details and examples.

Reviewer: always explain why the target _before_ linking to the target (easier skimming) 

```suggestion
For more information, refer to [File-based user authentication](/deploy-manage/users-roles/cluster-or-deployment-auth/file-based.md).
```

### Native realm

You can create custom users in the {{es}} native realm using {{es}} user management APIs or {{kib}}.

Refer to [Native user authentication](/deploy-manage/users-roles/cluster-or-deployment-auth/native.md) for more details.

### File realm

Custom users can also be created by providing the desired file realm content or a username and password in Kubernetes secrets, referenced in the {{es}} resource.

For more information, refer to [File-based user authentication](/deploy-manage/users-roles/cluster-or-deployment-auth/file-based.md).

### External authentication realms

You can also configure external authentication realms such as LDAP, OpenID Connect, or SAML in your ECK deployments by providing the appropriate {{es}} or {{kib}} configuration settings and any required [certificates or configuration files](/deploy-manage/deploy/cloud-on-k8s/custom-configuration-files-plugins.md).

For more information, refer to [External authentication](/deploy-manage/users-roles/cluster-or-deployment-auth/external-authentication.md).

## Creating custom roles

> Roles can be specified using the Role management API, or the Role management UI in {{kib}}. Additionally, file-based role management can be achieved by referencing Kubernetes secrets containing the roles specification.

Author: Perfect, makes sense! I'll rephrase that a bit :)

I just took the same approach as in the [original doc](https://www.elastic.co/guide/en/cloud-on-k8s/2.16/k8s-users-and-roles.html), but it's true that the original doc then provided all examples with secrets after the initial sentence.
